src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -0,0 +1,1150 @@
+        }
+
+
+        // IDbCommand.UpdateRowSource
+        static internal ArgumentOutOfRangeException InvalidUpdateRowSource(UpdateRowSource value)
+        {
+#if DEBUG
+            switch (value)
+            {
+                case UpdateRowSource.None:
+                case UpdateRowSource.OutputParameters:
+                case UpdateRowSource.FirstReturnedRecord:
+                case UpdateRowSource.Both:
+                    Debug.Assert(false, "valid UpdateRowSource " + value.ToString());
+                    break;
+            }
+#endif
+            return InvalidEnumerationValue(typeof(UpdateRowSource), (int)value);
+        }
+
+
+        //
+        // DbConnectionOptions, DataAccess
+        //
+        static internal ArgumentException ConnectionStringSyntax(int index)
+        {
+            return Argument(Res.GetString(Res.ADP_ConnectionStringSyntax, index));
+        }
+        static internal ArgumentException KeywordNotSupported(string keyword)
+        {
+            return Argument(Res.GetString(Res.ADP_KeywordNotSupported, keyword));
+        }
+        static internal ArgumentException InvalidMinMaxPoolSizeValues()
+        {
+            return ADP.Argument(Res.GetString(Res.ADP_InvalidMinMaxPoolSizeValues));
+        }
+        static internal ArgumentException ConvertFailed(Type fromType, Type toType, Exception innerException)
+        {
+            return ADP.Argument(Res.GetString(Res.SqlConvert_ConvertFailed, fromType.FullName, toType.FullName), innerException);
+        }
+
+
+        //
+        // DbConnection
+        //
+        static internal InvalidOperationException NoConnectionString()
+        {
+            return InvalidOperation(Res.GetString(Res.ADP_NoConnectionString));
+        }
+
+        static internal NotImplementedException MethodNotImplemented(string methodName)
+        {
+            NotImplementedException e = new NotImplementedException(methodName);
+            return e;
+        }
+
+        static private string ConnectionStateMsg(ConnectionState state)
+        {
+            switch (state)
+            {
+                case (ConnectionState.Closed):
+                case (ConnectionState.Connecting | ConnectionState.Broken):
+                    return Res.GetString(Res.ADP_ConnectionStateMsg_Closed);
+                case (ConnectionState.Connecting):
+                    return Res.GetString(Res.ADP_ConnectionStateMsg_Connecting);
+                case (ConnectionState.Open):
+                    return Res.GetString(Res.ADP_ConnectionStateMsg_Open);
+                case (ConnectionState.Open | ConnectionState.Executing):
+                    return Res.GetString(Res.ADP_ConnectionStateMsg_OpenExecuting);
+                case (ConnectionState.Open | ConnectionState.Fetching):
+                    return Res.GetString(Res.ADP_ConnectionStateMsg_OpenFetching);
+                default:
+                    return Res.GetString(Res.ADP_ConnectionStateMsg, state.ToString());
```

Comment:
We should be using the corefx resources mechanism for getting/formatting resource strings.